pkg/api/kubegraph/analysis/podspec_test.go

```diff
@@ -0,0 +1,38 @@
+package analysis
+
+import (
+	"testing"
+
+	osgraphtest "github.com/openshift/origin/pkg/api/graph/test"
+	kubeedges "github.com/openshift/origin/pkg/api/kubegraph"
+)
+
+func TestMissingSecrets(t *testing.T) {
+	g, _, err := osgraphtest.BuildGraph("../../../api/graph/test/bad_secret_refs.yaml")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	kubeedges.AddAllRequestedServiceAccountEdges(g)
+	kubeedges.AddAllMountableSecretEdges(g)
+	kubeedges.AddAllMountedSecretEdges(g)
+
+	if e, a := 1, len(FindMissingSecrets(g)); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+}
+
+func TestUnmountableSecrets(t *testing.T) {
+	g, _, err := osgraphtest.BuildGraph("../../../api/graph/test/bad_secret_refs.yaml")
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	kubeedges.AddAllRequestedServiceAccountEdges(g)
+	kubeedges.AddAllMountableSecretEdges(g)
+	kubeedges.AddAllMountedSecretEdges(g)
+
+	if e, a := 2, len(FindUnmountableSecrets(g)); e != a {
```

Comment:
Might be a stronger test if you verified that the markers were correctly linked to related secrets.